src/Pim/Bundle/EnrichBundle/Controller/ProductRestController.php

```
@@ -0,0 +1,350 @@
+     * @param ValidatorInterface        $validator
+     * @param UserContext               $userContext
+     * @param ObjectFilterInterface     $objectFilter
+     * @param CollectionFilterInterface $productEditDataFilter
+     * @param ProductBuilderInterface   $productBuilder
+     */
+    public function __construct(
+        ProductManager $productManager,
+        AttributeManager $attributeManager,
+        ProductUpdaterInterface $productUpdater,
+        SaverInterface $productSaver,
+        NormalizerInterface $normalizer,
+        ValidatorInterface $validator,
+        UserContext $userContext,
+        ObjectFilterInterface $objectFilter,
+        CollectionFilterInterface $productEditDataFilter,
+        ProductBuilderInterface $productBuilder
+    )
+    {
+        $this->productManager        = $productManager;
+        $this->attributeManager      = $attributeManager;
+        $this->productUpdater        = $productUpdater;
+        $this->productSaver          = $productSaver;
+        $this->normalizer            = $normalizer;
+        $this->validator             = $validator;
+        $this->userContext           = $userContext;
+        $this->objectFilter          = $objectFilter;
+        $this->productEditDataFilter = $productEditDataFilter;
+        $this->productBuilder        = $productBuilder;
+    }
+
+    /**
+     * Edit product
+     *
+     * @param int $id
+     *
+     * @Template("PimEnrichBundle:Product:edit.html.twig")
+     * @AclAncestor("pim_enrich_product_index")
+     *
+     * @return array
+     */
+    public function editAction($id)
+    {
+        return [
+            'productId' => $id
+        ];
+    }
+
+    /**
+     * @param string $id Product id
+     *
+     * @throws NotFoundHttpException If product is not found or the user cannot see it
+     *
+     * @return JsonResponse
+     */
+    public function getAction($id)
+    {
+        $product  = $this->findProductOr404($id);
```

If the product exists but the user can't see it we should throw a 401 error, not a 404.